### PR TITLE
Share m2 dependencies within a build scope

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Comment by aduprat** [comment]
When those volumes will be deleted?

**Comment by chibenwa** [comment]
At the end of the build (same step than image/container cleanup @aduprat )

**Review by rouazana — COMMENTED** [review]
I really doubt that the tmpfs trick is helping in performance. Can you check please?

**Comment by chibenwa** [comment]
> I really doubt that the tmpfs trick is helping in performance. Can you check please?

Let's just apply the first one then

**Review by rouazana — COMMENTED** [review]
Other question: what about the documented maven cache docker image? It's not supposed to do this job?

**Comment by chibenwa** [comment]
We no longer use it: it broke build isolation.

@mbaechler removed it, but I believe we can share it within a build scope.

File: workflow-job [hunk 1]
```diff
@@ -538,7 +538,7 @@ flows["${sha1}"] = {
                 rabbitMQProductExists = fileExists 'dockerfiles/run/guice/cassandra-rabbitmq-ldap'
                 memoryProductExists = fileExists 'dockerfiles/run/guice/memory'
 
-                sh "docker volume create ${m2volume}"
+                sh "docker volume create --driver local --opt type=tmpfs --opt device=tmpfs --opt o=size=400m ${m2volume}"
```
Comment (rouazana) [inline workflow-job]: hey, you sneakily put it back :trollface:

**Review by rouazana — COMMENTED** [review]
you should also remove DockerFileMavenCache file

File: workflow-job (hunk 1 above)
Comment (chibenwa) [reply workflow-job]: Arghh

**Review by mbaechler — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Review by rouazana — APPROVED** [review]

**Review by aduprat — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**